src/scripts/DataLayerListenerManagerFactory.js

```diff
@@ -37,73 +38,76 @@ ListenerManagerFactory.create = function(dataLayerManager) {
    */
   const ListenerManager = {
     /**
-     * Registers a listener based on a listener on item.
+     * Registers a listener.
      *
      * @function
-     * @param {Item} listenerOn The listener on.
+     * @param {Listener} listener The listener to register.
      */
-    register: function(listenerOn) {
-      const event = listenerOn.config.on;
+    register: function(listener) {
+      const event = listener.event;
 
       if (Object.prototype.hasOwnProperty.call(_listeners, event)) {
-        if (!_isRegistered(listenerOn)) {
-          _listeners[event].push(listenerOn);
+        if (_indexOf(listener) === -1) {
+          _listeners[listener.event].push(listener);
         }
       } else {
-        _listeners[event] = [listenerOn];
+        _listeners[listener.event] = [listener];
       }
     },
     /**
-     * Unregisters a listener based on a listener off item.
+     * Unregisters a listener.
      *
      * @function
-     * @param {Item} listenerOff The listener off.
+     * @param {Listener} listener The listener to unregister.
      */
-    unregister: function(listenerOff) {
-      const indexes = _getListenersMatchingListenerOff(listenerOff);
-      const event = listenerOff.config.off;
+    unregister: function(listener) {
+      const event = listener.event;
 
-      for (let i = indexes.length - 1; i > -1; i--) {
-        if (indexes[i] > -1) {
-          _listeners[event].splice(indexes[i], 1);
+      if (Object.prototype.hasOwnProperty.call(_listeners, event)) {
+        if (!listener.handler) {
+          _listeners[event] = [];
+        } else {
+          const index = _indexOf(listener);
+          if (index > -1) {
+            _listeners[event].splice(index, 1);
+          }
         }
       }
     },
     /**
-     * Triggers all the registered listeners matching the item.
+     * Triggers events related to the passed item.
      *
      * @function
-     * @param {Item} item The item.
+     * @param {Item} item The item for which to trigger events.
      */
-    triggerListeners: function(item) {
+    triggerEvents: function(item) {
```

Comment:
Shouldn't we call it `triggerHandlers` instead?

Reply:
I went with `triggerEvents`, because we're triggering all events associated with the item.